recipes/LibriSpeech/ASR/transformer/hparams/train_whisper_lora.yaml

```diff
@@ -0,0 +1,173 @@
+
+############################## Training Parameters #############################
+freeze_encoder: True
+number_of_epochs: 1
+weight_decay: 0.01
+lr_whisper: 1e-5
+warmup_steps: 500
+max_grad_norm: 2.0
+sorting: ascending
+precision: fp16 # bf16, fp16 or fp32
+eval_precision: fp16
+sampling_rate: 16_000
+
+# With data_parallel batch_size is split into N jobs
+# With DDP batch_size is multiplied by N jobs
+# This setup works well with 1x 32GB GPU
+batch_size: 2
+test_batch_size: 16
+grad_accumulation_factor: 8
+
+# Decoding parameters
+min_decode_ratio: 0.0
+max_decode_ratio: 1.0
+test_beam_size: 8
+
+# Lora configuration
+lora_rank: 16
+
+####################### Model Parameters #######################################
+
+train_loader_kwargs:
+    batch_size: !ref <batch_size>
+
+valid_loader_kwargs:
+    batch_size: !ref <test_batch_size>
+
+test_loader_kwargs:
+    batch_size: !ref <test_batch_size>
+
+
+epoch_counter: !new:speechbrain.utils.epoch_loop.EpochCounter
+    limit: !ref <number_of_epochs>
+
+############################## Augmentations ###################################
+
+# UNCOMMENT THIS SECTION TO ADD AUGMENTATIONS
+# speed_perturb: !new:speechbrain.augment.time_domain.SpeedPerturb
+#     orig_freq: !ref <sample_rate>
+#     speeds: [95, 100, 105]
+
+# # Frequency drop: randomly drops a number of frequency bands to zero.
+# drop_freq: !new:speechbrain.augment.time_domain.DropFreq
+#     drop_freq_low: 0  # Min frequency band dropout probability
+#     drop_freq_high: 1  # Max frequency band dropout probability
+#     drop_freq_count_low: 1  # Min number of frequency bands to drop
+#     drop_freq_count_high: 3  # Max number of frequency bands to drop
+#     drop_freq_width: 0.05  # Width of frequency bands to drop
+
+# # Time drop: randomly drops a number of temporal chunks.
+# drop_chunk: !new:speechbrain.augment.time_domain.DropChunk
+#     drop_length_low: 1
+#     drop_length_high: 5
+#     drop_count_low: 1000
+#     drop_count_high: 2000
+
+# # Augmenter: Combines previously defined augmentations to perform data augmentation
+# wav_augment: !new:speechbrain.augment.augmenter.Augmenter
+#     concat_original: True
+#     min_augmentations: 3
+#     max_augmentations: 3
+#     augment_prob: 1.0
+#     augmentations: [
+#         !ref <speed_perturb>,
+#         !ref <drop_freq>,
+#         !ref <drop_chunk>]
+
+############################## Models ##########################################
+
+whisper_pretrained: !new:speechbrain.lobes.models.huggingface_transformers.whisper.Whisper
+    source: !ref <whisper_hub>
+    freeze_encoder: !ref <freeze_encoder>
+    save_path: !ref <whisper_folder>
+    language: "english"
+    task: "transcribe"
+    sampling_rate: !ref <sampling_rate>
+
+whisper: !new:speechbrain.nnet.adapters.AdaptedModel
+    model_to_adapt: !ref <whisper_pretrained>
+    adapter_class: !name:speechbrain.nnet.adapters.LoRA
+    rank: !ref <lora_rank>
+    target_layers: [all-linear]
```

Comment:
We will need to be carreful about such a naming for a variable as it does not explicitely target elements, it could become hard to follow (if we want to target a single layer in the network?

Reply:
I don't know what would be the best tbf, hum... I like the boolean approach I think.

Reply:
It is on the same level right now. Another option would be instead of `**kwargs` add an `adapter_kwargs` argument with dictionary params to pass to the adapter layer.

Reply:
Like:

```yaml
whisper: !new:speechbrain.nnet.adapters.AdaptedModel
  model_to_adapt: !ref <whisper_pretrained>
  adapter_class: !name:speechbrain.nnet.adapters.LoRA
  all_linear: True
  adapter_kwargs:
    rank: !ref <lora_rank>
    scale: !ref <lora_scale>
```